packages/gatsby-plugin-page-creator/src/__tests__/path-utils.ts

```diff
@@ -0,0 +1,20 @@
+import { findCollectionPageFiles } from "../path-utils"
+import path from "path"
+
+describe(`findCollectionPageFiles`, () => {
+  it(`will find all and only the collection routes in a fixture`, async () => {
+    expect(
+      await findCollectionPageFiles(
+        path.join(__dirname, `fixtures`, `page-utils`, `collection-routes`)
+      )
+    ).toMatchInlineSnapshot(`
+      Array [
+        "{Other.extension}.js",
+        "{RootItem.name}.tsx",
+        "nested/{NestedItem.name}.tsx",
+        "nested/{NestedDirectoryOnlyItem}/edit.tsx",
+        "nested/{NestedDirectoryOnlyItem}/index.tsx",
+      ]
+    `)
+  })
+})
```

Comment:
This new test uses a fixture to test `findCollectionPageFiles` directly. The fixture files are blank because their content doesn't matter for this purpose, but I could easily see it expanded to be used for other logic in this plugin.